ui/src/app/cohort-search/cohort-search/cohort-search.component.tsx

```diff
@@ -155,24 +168,53 @@ export const CohortSearch = withCurrentCohortSearchContext()(class extends React
         this.setState({
           selectedIds: newSelections.map(s => s.parameterId),
           selections: newSelections
-        });
+        }, () => this.setUnsavedChanges());
       }
     });
   }
 
   componentWillUnmount() {
     this.subscription.unsubscribe();
     currentCohortCriteriaStore.next(undefined);
+    localStorage.removeItem(LOCAL_STORAGE_KEY_COHORT_CONTEXT);
   }
 
   closeSearch() {
     currentCohortSearchContextStore.next(undefined);
-    currentCohortCriteriaStore.next(undefined);
     // Delay hiding attributes page until sidebar is closed
     setTimeout(() => attributesSelectionStore.next(undefined), 500);
   }
 
+  setUnsavedChanges() {
+    const {cohortContext: {groupId, item, role}} = this.props;
+    const {selections} = this.state;
+    let unsavedChanges;
```

Comment:
can this be like :
_let unsavedChanges = selections.length > 0;
    if (groupId) {
      const requestItem = getItemFromSearchRequest(groupId, item.id, role);
      if (requestItem) {
        const sortAndStringify = (params) => JSON.stringify(params.sort((a, b) => a.id - b.id));
        unsavedChanges = sortAndStringify(requestItem.searchParameters) !== sortAndStringify(selections);
      }
    }_

Reply:
Good call, much cleaner this way. Changed.